src/pages/studyView/addChartButton/genericAssaySelection/GenericAssaySelection.tsx

```diff
@@ -29,35 +38,59 @@ export default class GenericAssaySelection extends React.Component<
     constructor(props: any) {
         super(props);
         makeObservable(this);
+        if (this.props.initialGenericAssayEntityIds) {
+            this._selectedGenericAssayEntityIds = this.props.initialGenericAssayEntityIds;
+        }
     }
     @observable private _selectedProfileOption:
         | (MolecularProfileOption & {
               profileName: string;
           })
+        | ISelectOption
         | undefined = undefined;
 
     @observable.ref private _selectedGenericAssayEntityIds: string[] = [];
     @observable private _genericAssaySearchText: string = '';
 
     @action.bound
-    private onAddChart() {
+    private onSubmit() {
         if (this.selectedProfileOption !== undefined) {
-            const charts = this._selectedGenericAssayEntityIds.map(entityId => {
-                return {
-                    name:
-                        entityId +
-                        ': ' +
-                        this.selectedProfileOption!.profileName,
-                    description: this.selectedProfileOption!.description,
-                    profileType: this.selectedProfileOption!.value,
-                    genericAssayType: this.props.genericAssayType,
-                    dataType: this.selectedProfileOption!.dataType!,
-                    genericAssayEntityId: entityId,
+            if (this.props.studyViewMode) {
+                const option = this
+                    .selectedProfileOption as MolecularProfileOption & {
+                    profileName: string;
                 };
-            });
-            this.props.onSubmit(charts);
-            // clear selected entities after submit new charts
-            this.clearSelectedEntities();
+                const charts = this._selectedGenericAssayEntityIds.map(
+                    entityId => {
+                        return {
+                            name: entityId + ': ' + option.profileName,
+                            description: option.description,
+                            profileType: option.value,
+                            genericAssayType: this.props.genericAssayType,
+                            dataType: option.dataType,
+                            genericAssayEntityId: entityId,
+                        };
+                    }
+                );
+                this.props.onSubmit(charts);
+                // clear selected entities after submit new charts
+                this.clearSelectedEntities();
+            } else {
+                const option = this.selectedProfileOption as ISelectOption;
+                // select profile if onSelectGenericAssayProfile exists
+                this.props.onSelectGenericAssayProfile &&
```

Comment:
it's kind of weird that we pass in 2 handlers (onSelectGenericAssayProfile and onSubmit).    Is there a reason?